websocket.go

```diff
@@ -253,25 +252,37 @@ func (p *WebSocketProxy) Start() {
 					continue
 				}
 
-				for client := range p.subscribers[swapID] {
-					// Marshal the message into JSON ([]byte)
-					messageBytes, err := json.Marshal(msg)
-					if err != nil {
-						log.Printf("Failed to marshal message for client: %v", err)
-						continue
-					}
+				// Marshal the message into JSON ([]byte)
+				messageBytes, err := json.Marshal(msg)
+				if err != nil {
+					log.Printf("Failed to marshal message for client: %v", err)
+					continue
+				}
 
-					// Write the message to the client WebSocket
-					err = client.Write(context.Background(), websocket.MessageText, messageBytes)
-					if err != nil {
-						log.Printf("Failed to send message to client: %v", err)
-						client.CloseNow()
-						delete(p.clients, client)
+				// Add clients to notify
+				for client := range p.subscribers[swapID] {
+					clientsToNotify[client] = messageBytes
```

Comment:
Is it possible a client would be notified about multiple swaps in a single loop here? In that case the message could be overwritten here.